js/__tests__/stats-test.js

```diff
@@ -0,0 +1,457 @@
+    var splitPictures = Stats.split({
+      "numGroups": 2, 
+      "numPictures": 3, 
+      "targetRating": 4.5, 
+      "pictures": mockPictures3, 
+      "splitFunc": "rr"
+    });
+
+    var expected = [
+      [
+        new Picture(4, "Picture 4"),
+        new Picture(3, "Picture 3"),
+        new Picture(2, "Picture 2")
+      ],
+      [
+        new Picture(5, "Picture 5"),
+        new Picture(6, "Picture 6"),
+        new Picture(7, "Picture 7")
+      ]
+    ];
+
+    expect(splitPictures).toEqual(expected);
+  });
+
+  it("round robin splits the data set into 3 groups", function() {
+    var splitPictures = Stats.split({
+      "numGroups": 3, 
+      "numPictures": 3, 
+      "targetRating": 4.5, 
+      "pictures": mockPictures3, 
+      "splitFunc": "rr"
+    });
+
+    var expected = [
+      [
+        new Picture(4, "Picture 4"),
+        new Picture(6, "Picture 6"),
+        new Picture(1, "Picture 1")
+      ],
+      [
+        new Picture(5, "Picture 5"),
+        new Picture(2, "Picture 2"),
+        new Picture(8, "Picture 8")
+      ],
+      [
+        new Picture(3, "Picture 3"),
+        new Picture(7, "Picture 7"),
+        new Picture(0, "Picture 0")
+      ]
+    ];
+
+    expect(splitPictures).toEqual(expected);
+  });
+});
+
+
+
+
+// Tests for our split algorithm (RANDOM)
+describe("random split", function() {
+  it("roud robin splits the data set", function() {
+    var splitPictures = Stats.split({
+      "numGroups": 2, 
+      "numPictures": 1, 
+      "targetRating": 9, 
+      "pictures": mockPictures2, 
+      "splitFunc": "rr"
+    });
+
+    expect(splitPictures.length).toBe(2);
+    _.each(splitPictures, function(group) {
+      expect(group.length).toBe(1);
+    });
+  });
+
+  it("also random splits the data set", function() {
+    var splitPictures = Stats.split({
+      "numGroups": 2, 
+      "numPictures": 1, 
+      "targetRating": 4.5, 
+      "pictures": mockPictures3, 
+      "splitFunc": "rr"
+    });
+
+    expect(splitPictures.length).toBe(2);
+    _.each(splitPictures, function(group) {
+      expect(group.length).toBe(1);
+    });
+  });
+
+  it("random splits the data set with more pictures", function() {
+    var splitPictures = Stats.split({
+      "numGroups": 2, 
+      "numPictures": 3, 
+      "targetRating": 4.5, 
+      "pictures": mockPictures3, 
+      "splitFunc": "rr"
+    });
+
+    expect(splitPictures.length).toBe(2);
+    _.each(splitPictures, function(group) {
+      expect(group.length).toBe(3);
+    });
+  });
+
+  it("random splits the data set into 3 groups", function() {
+    var splitPictures = Stats.split({
+      "numGroups": 3, 
+      "numPictures": 3, 
+      "targetRating": 4.5, 
+      "pictures": mockPictures3, 
+      "splitFunc": "rr"
+    });
+
+    expect(splitPictures.length).toBe(3);
+    _.each(splitPictures, function(group) {
+      expect(group.length).toBe(3);
+    });
+  });
+});
+
+
+
+
+
```

Comment:
not implemented yet, easier to see the different test blocks in the project contour lol